org.eclipse.m2e.profiles.core.tests/src/org/eclipse/m2e/profiles/core/internal/management/MavenProfileManagerTest.java

```diff
@@ -16,30 +16,43 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
-import java.io.File;
 import java.util.Arrays;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
-
 import org.eclipse.core.resources.IProject;
 import org.eclipse.m2e.core.MavenPlugin;
 import org.eclipse.m2e.core.internal.IMavenConstants;
 import org.eclipse.m2e.core.project.IMavenProjectFacade;
 import org.eclipse.m2e.profiles.core.internal.IProfileManager;
-import org.eclipse.m2e.profiles.core.internal.MavenProfilesCoreActivator;
 import org.eclipse.m2e.profiles.core.internal.ProfileData;
 import org.eclipse.m2e.tests.common.AbstractMavenProjectTestCase;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.util.tracker.ServiceTracker;
 
 
 public class MavenProfileManagerTest extends AbstractMavenProjectTestCase {
 
-  private final IProfileManager profileManager = MavenProfilesCoreActivator.getDefault().getProfileManager();
+	private static ServiceTracker<IProfileManager, IProfileManager> profileManagerTracker;
+
+	@BeforeClass
+	public static void setUpProfileManagerTracker() throws BundleException {
+		BundleContext context = FrameworkUtil.getBundle(MavenProfileManagerTest.class).getBundleContext();
```

Comment:
This plug-ins manifest specifies `Bundle-ActivationPolicy: lazy` so the `BundleContext` should never be null here.
Furthermore I think this code will never run outside of OSGi so I also expect that `getBundle()` never returns null unless the entire framework is in a weird state.

Reply:
For testing purpose this should be fine.

Reply:
Ups, I actually intended to add this comment in the class below.